components/previews/AnnouncementPreview.tsx

```diff
@@ -1,15 +1,16 @@
 import { IAnnouncement } from "../../@types/generated/contentful";
 
 type AnnouncementPreviewProps = {
-    entry: IAnnouncement;
-}
+  entry: IAnnouncement;
+};
 
 const AnnouncementPreview: React.FC<AnnouncementPreviewProps> = ({ entry }) => {
-    return (
-        <>
-            {entry.fields.title} at slug {entry.fields.slug}
-        </>
-    );
+  return (
+    <p>
+      {entry.fields.title} at slug{" "}
+      <a href={`/announcements/${entry.fields.slug}`}>{entry.fields.slug}</a>
```

Comment:
In Next.js, prefer using `Link` from [next/link](https://nextjs.org/docs/api-reference/next/link) over `a` tags. It will perform pre-fetching, navigate without reloading, and provides some nice props we can use.
```suggestion
      <Link href={`/announcements/${entry.fields.slug}`}>{entry.fields.slug}</Link>
```